### DISCUSSION TITLE
How to binding innerHTML?

### DISCUSSION THREAD

**Discussion by mzaini30** [alpinejs/alpine#1564] [1. Help]
In Svelte, I can binding innerHTML with this:

```html
<script>
 let content
</script>

<div contenteditable bind:innerHTML={content}></div>
```

How to do like that in Alpine JS? Thanks...

**Comment by danddanddand** [comment]
I think you might just be looking for x-html which was removed in v3. You can add the directive back yourself (https://alpinejs.dev/upgrade-guide#no-x-html) or wait for v3 toolkit to be released (https://github.com/alpine-collective/toolkit/blob/main/packages/x-html/src/index.js)

**Reply by calebporzio** [reply]
Good references.

If you want to replicate the behavior of `x-html` using existing Alpine (V3) APIs, this is what I would do:

```html
<div x-data="{ content: '<h1>some html</h1>' }">
    <div x-effect="$el.innerHTML = content">
    </div>
</div>
```

`x-effect="$el.innerHTML = content"` is the functional equivalent of `x-html="content"`

If you want to simply use `x-html`, like @danddanddand said, you can add it yourself pretty easily:
```html
<script>
    document.addEventListener('alpine:init', () => {
        Alpine.directive('html', (el, { expression }, { evaluateLater, effect }) => {
            let getHtml = evaluateLater(expression)

            effect(() => {
                getHtml(html => {
                    el.innerHTML = html
                })
            })
        })
    })
</script>
```

**Comment by earthboundkid** [comment]
FTR, x-html was added back in Alpine 3.1.
